spark/src/main/scala/org/apache/comet/CometExecIterator.scala

```diff
@@ -63,9 +64,28 @@ class CometExecIterator(
   }.toArray
   private val plan = {
     val conf = SparkEnv.get.conf
-    // Only enable unified memory manager when off-heap mode is enabled. Otherwise,
-    // we'll use the built-in memory pool from DF, and initializes with `memory_limit`
-    // and `memory_fraction` below.
+
+    val offHeapMode = CometSparkSessionExtensions.isOffHeapEnabled(conf)
+    val memoryLimit = if (offHeapMode) {
+      // in unified mode we share off-heap memory with Spark
+      ByteUnit.MiB.toBytes(conf.getSizeAsMb("spark.memory.offHeap.size"))
```

Comment:
Actually this memory limit number is not used, right?

Reply:
Hmm, okay. The pool size is only used in the following check, although it looks unnecessary. The callback should already do such check. Maybe we can remove it to make two unified memory pools consistent.

```rust
 let limit = self.pool_size.checked_div(num).unwrap();
 let size = reservation.size();
 if limit < size + additional {
   return resources_err!(
     "Failed to acquire {additional} bytes where {size} bytes already reserved and the fair limit is {limit} bytes, {num} registered"
    );
 }
```

Reply:
@kazuyukitanimura Should `FairUnified` be using `pool_size` or `pool_size_per_task`?

Reply:
I think the limit here is a subset of the overall pool size because it is divided by the number of tasks here:

```
let limit = self.pool_size.checked_div(num).unwrap();
```

Reply:
Oh oh, right. I missed it.

Reply:
But honestly I'm not sure if it makes sense because the overall pool size is not totally for Comet. If it is not really fair to all tasks as it depends on memory usage of Spark.